### PR TITLE
Add support for Logging e2e testcases

### PR DESCRIPTION
**Pull request by Jeevaann**

### CODE REVIEW

**Comment by Jeevaann** [comment]
> Hi @Jeevaann, Can you please squash the commits? And these commit contains changes for playbooks/roles/ocp-e2e/tasks/main.yml.
> 
> Also please add new line at end of the files.

Yes, I have squashed the commits and also added new line at the end of the files. This is all together a new playbook which will run logging e2e testcases from openshift-test-private repo.

File: playbooks/roles/ocp-e2e/tasks/main.yml
Comment (varad-ahirwadkar) [inline playbooks/roles/ocp-e2e/tasks/main.yml:1]: Is this file modified ?

File: playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    sed -e 's/\"/\\"/g;s/.*/\"&\"/'  {{ results_dir }}/logging_tests_private_e2e_output.txt   | awk '/Failing tests:/,EOF' | tail -n +3 | head -n -4 > {{ results_dir }}/failed-logging-tests-private-e2e-results.txt
+  args:
+    chdir: "{{ test_dir }}"
+  environment: "{{ logging_e2e_env }}"
+  when: logging_e2e_testcases == "all" or logging_e2e_testcases == ""
+  register: opp
+
+- name: Prepare test suites and run critical e2e tests
+  shell: |
+    ./bin/extended-platform-tests run all --dry-run | grep -E "sig-openshift-logging.*Critical" > logging_tests_critical.txt
```
Comment (varad-ahirwadkar) [inline playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml:125]: Critical test cases should also be included in the logging tests, right? Are these redundant ?

File: playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    ./bin/extended-platform-tests run -f ./logging_tests_critical.txt {{ repository_command }} -o {{ results_dir }}/logging_tests_critical_output.txt
+    sed -e 's/\"/\\"/g;s/.*/\"&\"/'  {{ results_dir }}/logging_tests_private_e2e_output.txt   | awk '/Failing tests:/,EOF' | tail -n +3 | head -n -4 > {{ results_dir }}/failed-logging-tests-private-e2e-results.txt
+  args:
+    chdir: "{{ test_dir }}"
+  environment: "{{ logging_e2e_env }}"
+  when: logging_e2e_testcases == "critical"
+  register: opp
+
```
Comment (varad-ahirwadkar) [inline playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml:133]:
```suggestion
  register: opp
 
```

File: playbooks/roles/ocp-logging-tests-private-e2e/vars/main.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+
+#var file for logging e2e 
+logging_e2e_env:
+   PATH: "/usr/local/go/bin:{{ ansible_env.PATH }}"
+   KUBECONFIG: "{{ ansible_env.HOME }}/.kube/config"
+
```
Comment (varad-ahirwadkar) [inline playbooks/roles/ocp-logging-tests-private-e2e/vars/main.yml:7]:
```suggestion
  
```

File: playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml (hunk 1 above)
Comment (Jeevaann) [reply playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml:125]: Yes, here we can choose between Critical logging testcases or all the logging related testcases. Critical testcases will be part of all logging testcases.

File: playbooks/roles/ocp-e2e/tasks/main.yml
Comment (Jeevaann) [reply playbooks/roles/ocp-e2e/tasks/main.yml:1]: No, I haven't modified this file

File: examples/all.yaml [hunk 4]
```diff
@@ -382,4 +382,11 @@ coo_namespace: "openshift-observability-operator"
...
- 
\ No newline at end of file
+ 
+# ocp-logging-tests-private-e2e vars
+ocp_logging_tests_private_e2e_enabled: false #Set to true to enable logging e2e tests from openshift-tests-private repo
+ocp_logging_tests_private_e2e_dir: /tmp/ocp_logging_tests_private/ #Test directory path
+ocp_logging_tests_private_e2e_git: "https://github.com/openshift/openshift-tests-private/" #Git repo url for the e2e tests
+ocp_logging_tests_private_e2e_git_branch: "" #Git repo branch for e2e tests
+golang_tarball_logging_e2e: "https://dl.google.com/go/go1.22.6.linux-ppc64le.tar.gz" #HTTP URL for golang tarball
+logging_e2e_testcases: "" #Set to either all or critical
```
[the remark this answers is not in the document]
Comment (Jeevaann) [reply examples/all.yaml]: Added new line at the end

File: playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml (hunk 2 above)
Comment (Jeevaann) [reply playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml]: Removed space and added new line

File: playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml (hunk 1 above)
Comment (varad-ahirwadkar) [reply playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml:125]: ok.

File: playbooks/roles/ocp-e2e/tasks/main.yml
Comment (varad-ahirwadkar) [inline playbooks/roles/ocp-e2e/tasks/main.yml:1]: can you please remove this file from commit?

Comment (Jeevaann) [reply playbooks/roles/ocp-e2e/tasks/main.yml:1]: Removed the playbooks/roles/ocp-e2e/tasks/main.yml file from commit.

File: playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,133 @@
+---
+
+# tasks file for ocp4 e2e
+- name: Install required packages
+  package:
+    name:
+      - git
+    state: latest
+
+- name: install the 'Development tools' package group
```
Comment (varad-ahirwadkar) [inline playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml:10]:
```suggestion
- name: Install the 'Development tools' package group
```

File: playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+  package: name="@Development tools" state=latest
+  when:  ansible_distribution == 'CentOS' or
+         ansible_distribution == 'RedHat' or
+         ansible_distribution == 'Fedora'
+
+- name: install the 'Build Essential' package group
```
Comment (varad-ahirwadkar) [inline playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml:16]:
```suggestion
- name: Install the 'Build Essential' package group
```

File: playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    version: "{{ ocp_logging_tests_private_e2e_git_branch }}"
+
+- name: Run make build command at target
+  shell: make
+  environment: "{{ logging_e2e_env }}"
+  args:
+    chdir: "{{ test_dir }}"
+
+- name: Check if the binary is created
+  shell: ls -hl {{ test_dir }}/bin/extended-platform-tests | wc -l
```
Comment (varad-ahirwadkar) [inline playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml:57]: Can we use [stat_module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/stat_module.html) to check the binary ?

File: playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+  file:
+    path: "{{ results_dir }}"
+    state: "{{ item }}"
+    mode: '0755'
+  with_items:
+  - absent
+  - directory
+
+# Check Nodes status
+- name: Check all nodes are Ready
```
Comment (varad-ahirwadkar) [inline playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml:80]: Can we use this role [check-cluster-health](https://github.com/ocp-power-automation/ocp4-playbooks-extras/blob/main/playbooks/roles/check-cluster-health) ? It checks the cluster health.

File: examples/ocp_logging_tests_private_e2e_vars.yaml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+---
+
+ocp_logging_tests_private_e2e_enabled: false #Set to true to enable logging e2e tests from openshift-tests-private repo
```
Comment (varad-ahirwadkar) [inline examples/ocp_logging_tests_private_e2e_vars.yaml:3]: This variable is not being used. Please update this file [main.yml](https://github.com/ocp-power-automation/ocp4-playbooks-extras/blob/main/playbooks/main.yml).

File: playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml (hunk 8 above)
Comment (Jeevaann) [reply playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml]: Yes, used the role check-cluster-health

File: playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml (hunk 7 above)
Comment (Jeevaann) [reply playbooks/roles/ocp-logging-tests-private-e2e/tasks/main.yml]: yes, used stat_module to check the binary.

File: examples/ocp_logging_tests_private_e2e_vars.yaml (hunk 9 above)
Comment (Jeevaann) [reply examples/ocp_logging_tests_private_e2e_vars.yaml:3]: Imported the role in playbooks/main.yml.

File: playbooks/main.yml [hunk 10]
```diff
@@ -118,3 +118,6 @@
 
 - import_playbook: ocp-etcd-rotate-encryption-key.yml
   when: ocp_etcd_key_rotation_enable is defined and ocp_etcd_key_rotation_enable
+
+- import_playbook: ocp-logging-tests-private-e2e.yml
+  when: ocp_logging_tests_private_e2e_enabled
```
Comment (varad-ahirwadkar) [inline playbooks/main.yml:123]:
```suggestion
  when: ocp_logging_tests_private_e2e_enabled is defined and ocp_logging_tests_private_e2e_enabled
 
```

Comment (Jeevaann) [reply playbooks/main.yml]: Implemented the suggestion.

**Review by pravin-dsilva — APPROVED** [review]
/approve

**Pull request merged**